lib/default_value_for.rb

```diff
@@ -65,23 +65,18 @@ def default_value_for(attribute, options = {}, &block)
         allows_nil = opts.fetch('allows_nil', true)
       end
 
-      if !method_defined?(:set_default_values)
+      if !singleton_methods(false).include?(:_default_attribute_values)
         include(InstanceMethods)
 
         after_initialize :set_default_values
 
         class_attribute :_default_attribute_values
         class_attribute :_default_attribute_values_not_allowing_nil
 
-        extend(DelayedClassMethods)
-        init_hash = true
-      else
-        init_hash = !singleton_methods(false).include?(:_default_attribute_values)
-      end
-
-      if init_hash
         self._default_attribute_values = {}
         self._default_attribute_values_not_allowing_nil = []
```

Comment:
I'll need to dig into the history of the original code to better understand why we avoided initializing/resetting these class attributes previously and why it's ok now.  I see this passes on older rails versions but I'm not sure the reason why this code was written like this.

@k-tsuchiya-jp did you figure out why your change is safe to do for rails 6 through 8?